lib/chef/monkey_patches/win32/registry.rb

```diff
@@ -22,15 +22,20 @@
 module Win32
   class Registry
 
+    GEM_VERSION_RUBY = Gem::Version.new(RUBY_VERSION)
+    GEM_VERSION_217  = Gem::Version.new("2.1.7")
+    GEM_VERSION_220  = Gem::Version.new("2.2.0")
+    GEM_VERSION_223  = Gem::Version.new("2.2.3")
+
     module API
 
       extend Chef::ReservedNames::Win32::API::Registry
 
       module_function
 
-      if RUBY_VERSION =~ /^2\.1/
-        # ::Win32::Registry#delete_value is broken in Ruby 2.1 (up to Ruby 2.1.6).
-        # This should be resolved in a later release (see note #9 in link below).
+      if GEM_VERSION_RUBY < GEM_VERSION_217 || GEM_VERSION_RUBY == GEM_VERSION_220
```

Comment:
Unfortunately, this is still hopelessly broken.  It seems the core ruby folks simply don't try out the windows registry API at all.

See https://github.com/ruby/ruby/blob/trunk/ext/win32/lib/win32/registry.rb

As of Feb 2015, they import the correct functions now - https://github.com/ruby/ruby/commit/a6516ff5f7f0daab88b44c64c8bd9dfa5317d6d0

But DeleteKey and DeleteValue now call non-existent methods because they weren't updated to call RegDeleteKeyW or RegDeleteValueW.  That's why your AppVeyor tests are failing with undefined constants.